### PR TITLE
More optimization of AppVeyor config

### PR DESCRIPTION
**Pull request by chadwalters**
Only load git submodules when they are needed. Remove redundant .NET Core build.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -83,12 +83,18 @@
...
-
     install:
         - ps: >-
-            git submodule update --init --recursive
+            if (($env:BOND_BUILD -eq 'C++') -or ($env:BOND_BUILD -eq 'Python')) {
+
+                git submodule update --init thirdparty\rapidjson
+
+                # If gRPC is not disabled, init gRPC and its dependencies
+                if (($env:BOND_BUILD -eq 'C++') -and (-not ($env:BOND_CMAKE_FLAGS -match '-DBOND_ENABLE_GRPC=FALSE'))) {
```
Comment (chwarr) [inline appveyor.yml]: Minor: I think the lack of a blank line between the comment and the condition is OK because we're inside of another condition, but for robustness under future editing, I think we should have a blank line.

Comment (lalo) [inline appveyor.yml]: line 176 is also culprit of this (not part of this commit)

**Comment by chwarr** [comment]
Merged to grpc++ in commit 2be617f1.
